plugin/src/main/groovy/com/novoda/staticanalysis/StaticAnalysisExtension.groovy

```diff
@@ -60,4 +69,26 @@ class StaticAnalysisExtension {
         rules
     }
 
+    NamedDomainObjectContainer<Violations> getAllViolations() {
+        allViolations
+    }
+
+    ViolationsEvaluator getEvaluator() {
+        createEvaluator()
+    }
+
+    Logger getLogger() {
+        project.logger
+    }
+
+    void evaluator(Action<? super Set<Violations>> action) {
+        createEvaluator = {
```

Comment:
I don't understand why you need an extra `Closure` here? Is creating the internal evaluator really need to be lazy? Here, creating new ViolationsEvaluator and executing given action later with `evaluate` method already provides laziness, isn't it?

Reply:
I suppose that was because of part of the `Input` needed to be accessed lazily (ie: thresholds after configuration). In general I believe this will allow us to avoid to wonder about similar issues in the action provided, don't you think?

Reply:
ok, discussed IRL: the question was more around the need of having a closure to create the evaluator. The reason is that this way we can pass the closure (ie: factory) to the evaluation task instead of passing around the evaluator itself. Any thoughts @devisnik?

Reply:
I just confirmed that the lazy creation of the evaluator will allow us to access the correct configured properties from the plugin extension(eg: `reportUrlRenderer`)